applications/nrf_desktop/src/modules/hids.c

```diff
@@ -237,37 +236,56 @@ static int module_init(void)
 	size_t feat_pos = 0;
 
 	if (IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_MOUSE_SUPPORT)) {
+		static const u8_t mask[] = REPORT_MASK_MOUSE;
+		BUILD_ASSERT(sizeof(mask) == ceiling_fraction(REPORT_SIZE_MOUSE, 8));
+		BUILD_ASSERT(REPORT_ID_MOUSE < ARRAY_SIZE(report_index));
+
 		input_report[ir_pos].id       = REPORT_ID_MOUSE;
 		input_report[ir_pos].size     = REPORT_SIZE_MOUSE;
 		input_report[ir_pos].handler  = mouse_notif_handler;
-		input_report[ir_pos].rep_mask = mouse_mask;
+		input_report[ir_pos].rep_mask = mask;
 
 		report_index[input_report[ir_pos].id] = ir_pos;
 		ir_pos++;
 	}
 
 	if (IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_KEYBOARD_SUPPORT)) {
-		input_report[ir_pos].id      = REPORT_ID_KEYBOARD_KEYS;
-		input_report[ir_pos].size    = REPORT_SIZE_KEYBOARD_KEYS;
-		input_report[ir_pos].handler = keyboard_notif_handler;
+		static const u8_t mask[] = REPORT_MASK_KEYBOARD_KEYS;
+		BUILD_ASSERT(sizeof(mask) == ceiling_fraction(REPORT_SIZE_KEYBOARD_KEYS, 8));
```

Comment:
Keyboard, system control and consumer control will not work here

Reply:
Fixed